custom_components/ha-google-home/sensor.py

```diff
@@ -46,46 +60,62 @@ def device_state_attributes(self):
 class GoogleHomeAlarmSensor(GoogleHomeAlarmEntity):
     """Representation of a Sensor."""
 
-    def __init__(self, coordinator, entry, device):
+    def __init__(self, coordinator, entry, device_name, alarms):
         """Initialize the sensor."""
         super().__init__(coordinator, entry)
-        self._name = device.device_name
-        self._alarms = [format_alarm_information(alarm) for alarm in device.alarms]
+        self._name = device_name
 
-        if len(self._alarms) > 0:
-            self._state = self._alarms[0][FIRE_TIME_IN_S]
-        else:
-            self._state = STATE_OFF
+    @property
+    def state(self):
+        alarms = self.get_alarms_data()
+        state = alarms[0][LOCAL_TIME] if alarms else STATE_OFF
+        return state
 
     @property
     def device_state_attributes(self):
         """Return the state attributes."""
         return {
-            "alarms": self._alarms,
+            "alarms": self.get_alarms_data(),
             "device": str(self.name),
             "integration": DOMAIN,
         }
 
+    def get_alarms_data(self):
```

Comment:
you can mark it private
```suggestion
    def _get_alarms_data(self):
```